sdk/src/main/scala/com/horizen/block/SidechainsHashMap.scala

```diff
@@ -0,0 +1,155 @@
+package com.horizen.block
+
+
+import com.google.common.primitives.{Bytes}
+import com.horizen.utils.{ByteArrayWrapper, BytesUtils, MerklePath, MerkleTree, Utils}
+
+import scala.collection.JavaConverters._
+import scala.collection.mutable
+import scala.util.Try
+
+class SidechainHashList
+{
+  private var forwardTransfersHash: Option[Array[Byte]] = None
+  private var withdrawalCertificateHash: Option[Array[Byte]] = None
+
+  def setForwardTransfersHash(hash: Array[Byte]): Unit = {
+    forwardTransfersHash = Some(hash)
+  }
+
+  def setWithdrawalCertificateHash(hash: Array[Byte]): Unit = {
+    withdrawalCertificateHash = Some(hash)
+  }
+
+  def getTxsHash: Array[Byte] = {
+    val backwardMerkleRoot = SidechainHashList.MAGIC_SC_STRING
+    BytesUtils.reverseBytes(
+      Utils.doubleSHA256Hash(
+        Bytes.concat(
+          BytesUtils.reverseBytes(forwardTransfersHash.getOrElse(SidechainHashList.MAGIC_SC_STRING)),
+          BytesUtils.reverseBytes(backwardMerkleRoot)
+        )
+      )
+    )
+  }
+
+  def getSidechainHash(sidechainId: Array[Byte]): Array[Byte] = {
+    SidechainHashList.getSidechainHash(
+      sidechainId,
+      getTxsHash,
+      withdrawalCertificateHash.getOrElse(SidechainHashList.MAGIC_SC_STRING)
+    )
+  }
+
+  def getNeighbourProof(sidechainId: Array[Byte], merkleTree: MerkleTree): NeighbourProof = {
```

Comment:
move this method to SidechainsCommitmentTree and rename NeighbourProof to SidechainCommitmentEntryProof